src/vm/appdomain.cpp

```diff
@@ -12956,6 +12956,11 @@ void AppDomain::CreateADUnloadWorker()
         DWORD stackSize = 512 * 1024; // leave X64 unchanged since we have plenty of VM
 #endif
         Thread *pThread = SetupUnstartedThread();
+        if (pThread == NULL)
+        {
+            ThrowOutOfMemory();            
```

Comment:
It would be better if you created a SetupUnstartedThreadNoThrow for GC to use, then you don't need to modify all these other places and let new throw as it would instead of having to throw OOM in all the places that call it.